react/SearchWrapper.tsx

```diff
@@ -73,34 +74,53 @@ const getPageEventName = (
   return (mapEvent[category] || fallbackView) as PageEventName
 }
 
+const getDecodeURIComponent = (encodedURIComponent: string) => {
+  try {
+    return decodeURIComponent(encodedURIComponent)
+  } catch {
+    return encodedURIComponent
+  }
+}
+
 interface GetTitleTagParams {
   titleTag: string
   storeTitle: string
   term?: string
   pageTitle?: string
+  pageNumber?: number
+  removeStoreNameTitle?: boolean
 }
 
 const getTitleTag = ({
   titleTag,
   storeTitle,
   term,
   pageTitle,
+  pageNumber = 0,
+  removeStoreNameTitle,
 }: GetTitleTagParams) => {
+  const titleNumber = pageNumber > 0 ? ` #${pageNumber}` : ''
+  const storeTitleFormatted = removeStoreNameTitle ? '' : ` - ${storeTitle}`
+
   if (titleTag) {
-    try {
-      return `${decodeURIComponent(titleTag)} - ${storeTitle}`
-    } catch {
-      return `${titleTag} - ${storeTitle}`
-    }
+    return `${getDecodeURIComponent(
+      titleTag
+    )}${titleNumber}${storeTitleFormatted}`
```

Comment:
These changes will change the title for all account pages. Can you explain better the need for this change?

Reply:
Yeah, if we do it this way, there's no way for an account to opt-out of having this added to their title tags

Reply:
Yes, this will apply, but only if some conditions are fulfilled on the constants **titleNumber** or **storeTitleFormatted**.

**titleNumber** and **storeTitleFormatted** depend on the value of **enablePageNumberTitle** and **removeStoreNameTitle**, by default, the value of enablePageNumberTitle and removeStoreNameTitle is false.
if a store does not activate any of these two parameters in [Admin-pages](https://github.com/vtex-apps/admin-pages/pull/414), the title of the pages will not be affected, since the value of titleNumber and storeTitleFormatted would be an empty string and the title of the web page would be represented as up to now.

if there is any question, let me know

Reply:
@igorbrasileiro @victorhmp from a SEO perspective the meta title for the different PLP pages should not be exactly the same, otherwise it will be consider as duplicated. The solution is to add the page number to the title from the second onwards and in this way avoid any duplication.
As Christian explained the idea is to use a setting in order to not modify any default behaviour for all sites. The setting will be false by default, so this change will be applied only for sites that enable it.

Reply:
Would be good to leave a comment explaining that those changes only will happen if <some setting> is enabled with a specific value.

Reply:
done, I included the comment when declaring **titleNumber** and **storeTitleFormatted**